flytekit/core/workflow.py

```diff
@@ -746,14 +746,17 @@ def compile(self, **kwargs):
                     )
                 workflow_outputs = workflow_outputs[0]
             t = self.python_interface.outputs[output_names[0]]
-            b, _ = binding_from_python_std(
-                ctx,
-                output_names[0],
-                self.interface.outputs[output_names[0]].type,
-                workflow_outputs,
-                t,
-            )
-            bindings.append(b)
+            try:
+                b, _ = binding_from_python_std(
+                    ctx,
+                    output_names[0],
+                    self.interface.outputs[output_names[0]].type,
+                    workflow_outputs,
+                    t,
+                )
+                bindings.append(b)
+            except Exception as e:
+                raise AssertionError(f"Failed to bind output {output_names[0]} for function {self.name}: {e}") from e
```

Comment:
@wild-endeavor should these be AssertionErrors? or FlyteExceptions?

Reply:
I used `AssertionError` in analogy to how it is done [here](https://github.com/flyteorg/flytekit/blob/c559e2649f9ed0de02e9fe0151d491bb7dde664e/flytekit/core/promise.py#L1089) but I'm happy to change to `FlyteException` if you prefer.

Reply:
Let's do it like this https://github.com/flyteorg/flytekit/pull/2047/commits/50ac42fbde0f56cf55aa84251db1caf45b348ae5?